lisp/ess-inf.el

```diff
@@ -1321,16 +1284,7 @@ Example (ess-boolean-command \"2>1\n\")"
     (ess-kill-last-line)
     (buffer-substring (point-min) (point-max))))
 
-;; (ess-async-command "{cat(1:5);Sys.sleep(5);cat(2:6)}\n" nil (get-process "R")
```

Comment:
could you please move these commetns to inf test file?

Reply:
ess-async-command seems to return a buffer. Is that right?

Reply:
Yes, that's correct because `ess-command` returns the buffer. Sometimes consumers of `ess-command` need to investigate the output buffer. Less useful for async though.

Reply:
Doesn't seem to work for me:

```
(ess-async-command "{cat(1:3);Sys.sleep(1);cat(4:5)}\n" (get-buffer-create "test") nil
                         (lambda (proc string) (message "done")))
```

results in the "test" buffer getting "1 2 3" but not the "4 5".

Reply:
:) The test buffer returns immediately and there is only a bit of output there, for async commands that buffer should be used for errors or something, not for output. The output comes in the callback "string" argument. 

Try

```el
(ess-async-command "{cat(1:5);Sys.sleep(5);cat(2:6)}\n" nil (get-process "R")
                   (lambda (proc string) (message "done %s" string)))
```

You should get `done 2 3 4 5 6> ` message in 5 seconds. 

This async commands are not really async in R sense (obviously). It's async in the sense that it's non-blocking on elisp level, unlike ess-command. A poor man pathetic attempt ... but what can one do :/. I am using it in polymode though.

Reply:
Thanks. I was misunderstanding how the function worked. I just added a test. Will merge after making sure it doesn't break on travis.